capture/genBitmaps.js

```diff
@@ -169,10 +169,32 @@ function processScenario (casper, scenario, scenarioOrVariantLabel, scenarioLabe
       if (!scenario.hasOwnProperty('selectors')) {
         scenario.selectors = ['body'];
       }
+
+    // if similar selectors exist on the given page, expand and test them too
+    if (scenario.enableSelectorExpansion) {
+        scenario.selectors = scenario.selectors.reduce(function(acc, selector, index) { 
+            var expandedSelectors = casper.evaluate(function(selector, index) {
+                return [].slice.call(document.querySelectorAll(selector)).map(function(element, i) {
+
+                        // by using __n{index}__, we are making sure that this classname does not clash with user defined ones 
+                        var nPart = '__n' + index + '__' + i;
```

Comment:
This is used to identify the reference image, assume the reference image file name is something like
```
..btn-primary_0_desktop.png
```
After the expansion, it would look like
```
..btn-primary__n0__0_0_desktop.png
..btn-primary__n0__1_0_desktop.png
```

On line number 196, i reset the index to the actual reference file's. So that the script would compare the expanded selectors  against  ```..btn-primary_0_desktop.png```.